lib/fieldhand/paginator.rb

```diff
@@ -81,10 +80,14 @@ def request(query = {})
       request_uri.query = encode_query(query)
 
       logger.info('Fieldhand') { "GET #{request_uri}" }
-      http.get(request_uri.request_uri).body
+      res = http.get(request_uri.request_uri)
+      raise NetworkError.new("Invalid response: #{res.code} #{res.msg}", res) unless res.is_a?(::Net::HTTPSuccess)
+
+      request.body
```

Comment:
yes, I did forget renaming `request` to `res`! However I'd prefer to keep it as `res` because to avoid upsetting rubocop (line too long, that I could split on multiple lines but then I'd have to move the `unless` and ad that point the line wouldn't need splitting anyway but the method will likely have too many lines 😝 )